File: ios-mma-recomp/MMARecomp/Models/MMASession.swift
import Foundation

struct MMASession: Codable, Identifiable {
    var id: UUID
    var userId: UUID
    var date: String
    var wodContent: String
    var roundsSets: String?
    var ressenti: Int?
    var notesTechnique: String?
    var createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case date
        case wodContent = "wod_content"
        case roundsSets = "rounds_sets"
        case ressenti
        case notesTechnique = "notes_technique"
        case createdAt = "created_at"
    }
}

struct NewMMASession: Codable {
    var date: String
    var wodContent: String
    var roundsSets: String?
    var ressenti: Int?
    var notesTechnique: String?

    enum CodingKeys: String, CodingKey {
        case date
        case wodContent = "wod_content"
        case roundsSets = "rounds_sets"
        case ressenti
        case notesTechnique = "notes_technique"
    }
}

File: android-mma-recomp/app/src/main/java/com/example/mmarecomp/model/MmaSession.kt
package com.example.mmarecomp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MmaSession(
    val id: String,
    @SerialName("user_id") val userId: String,
    val date: String,
    @SerialName("wod_content") val wodContent: String,
    @SerialName("rounds_sets") val roundsSets: String? = null,
    val ressenti: Int? = null,
    @SerialName("notes_technique") val notesTechnique: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
)

@Serializable
data class NewMmaSession(
    val date: String,
    @SerialName("wod_content") val wodContent: String,
    @SerialName("rounds_sets") val roundsSets: String? = null,
    val ressenti: Int? = null,
    @SerialName("notes_technique") val notesTechnique: String? = null,
)
